### DISCUSSION TITLE
Disabling admin console ssl requirement

### DISCUSSION THREAD

**Discussion by ahamid** [keycloak/keycloak#12155] [Archive]
I am configuring a Keycloak 18 instance in edge proxy mode with an admin console without ssl requirement.

We have historically configured the realm via db to disable ssl requirement on the admin console, as [some](https://stackoverflow.com/a/56170579) [suggest](https://stackoverflow.com/a/42002397).

It appears there may be a way to configure this programmatically at runtime during startup, which would be preferable, however I'm not sure how to set the cited `quarkus.keycloak.ssl-required` property:

`http://localhost:9550/auth/admin/master/console/config`
Actual:
`{"realm":"master","auth-server-url":"https://localhost:9550/auth/","ssl-required":"external","resource":"security-admin-console","public-client":true,"confidential-port":0}`
Desired:
`{"realm":"master","auth-server-url":"http://localhost:9550/auth/","ssl-required":"none","resource":"security-admin-console","public-client":true,"confidential-port":0}`

https://github.com/quarkusio/quarkus/issues/2921#issuecomment-1135139066

Is this exclusively a property of the realm configuration, requiring manual reconfiguration, export and import?

**Comment by ahamid** [comment]
To follow up on my own issue, through trial and error it appears the crucial Keycloak setting is the environment variable `KC_HOSTNAME_STRICT_HTTPS=false`, irrespective of the ssl_required mode defaulting to `external`.

**Comment by omasseau** [comment]
I tried to set KC_HOSTNAME_STRICT_HTTPS=false but still https links are generated when trying to access the admin console (I'm behind a reverse-proxy) .


When using this config:
- KC_HTTP_ENABLED=true
- KC_PROXY=passthrough
- KC_HOSTNAME=localhost
- KC_HOSTNAME_PORT=80
- KC_HOSTNAME_STRICT=false
- KC_HOSTNAME_STRICT_HTTPS=false
- KC_HTTP_RELATIVE_PATH=/auth

and accessing the admin console I can see that Keycloak is calling this url which is wrong : 
https://localhost:80/auth/realms/master/protocol/openid-connect/3p-cookies/step1.html?version=7kl62

**Reply by ahamid** [reply]
Have you tried setting: `KC_HTTPS_ENABLED=false`?

**Reply by aovasylenko** [reply]
was it solved?

**Reply by rajeshkppt** [reply]
doesn't work with KC_HTTPS_ENABLED or KC_HOSTNAME_STRICT_HTTPS

**Comment by hansjakobli** [comment]
I have the same issue. Use the following environment variables in my dockerfile.
`ENV KC_HTTP_ENABLED=true
ENV KC_HTTP_PORT=9010
ENV KC_HOSTNAME_ADMIN_URL=http://localhost:9010
ENV KC_PROXY=edge`

When trying to access the console in the browser I notice that http://localhost:9010/admin/master/console/config returns the following:
`{"realm":"master","auth-server-url":"https://localhost/","ssl-required":"external","resource":"security-admin-console","public-client":true,"confidential-port":0}`

If I look at the client `security-admin-console` in the master realm, then 'Root URL' is by default `${authAdminUrl}`. Does this variable not respect the defined environment variables?

I see a similar issue here https://github.com/keycloak/keycloak/issues/12529

**Reply by hansjakobli** [reply]
...adjusting the `security-admin-console` 'Rood URL' does not seem to help. The /config endpoint remains unchanged.

**Reply by hansjakobli** [reply]
I think I finally understood. While the admin console can be exposed without the ssl (without proxy) the authentication flow still uses ssl (via proxy). It won't be possible to get the `auth-server-url` without the `https`.

If you need to change the `keycloak.ssl-required` property you can do this in the CLI with:
`./kcadm.sh config credentials --server http://localhost:9010/ --realm master --user admin`
and
`./kcadm.sh update realms/master -s sslRequired=NONE`

Changing sslRequired won't affect `auth-server-url` though.

In my case I used
```
ENV KC_HTTP_ENABLED=true
ENV KC_HTTP_PORT=9010
ENV KC_PROXY=edge
ENV KC_HOSTNAME=my.domain.com
```

**Reply by abhijiremaniche** [reply]
<img width="943" alt="image" src="https://user-images.githubusercontent.com/43595998/226278983-388117c8-69d3-4bd9-b981-35555afc2cab.png">


set required SSL  to None through admin console

**Comment by saalihou** [comment]
For me, it seems like `KC_HTTP_*` settings and the `KC_PROXY` setting were conflicting.
I have an ssl enabled AWS ELB proxy in front of Keycloak. I initially simply enabled http with `KC_HTTP_ENABLED=true`, `KC_HTTPS_ENABLED=false` and `KC_HOSTNAME=<my_host>` `KC_HOSTNAME_PORT=443`. 
When I added `KC_PROXY=edge` to the mix it did not work at first. But then I removed all HTTP/HTTPS related settings (keeping only `KC_HOSTNAME=<my_host>` and `KC_PROXY=edge` and that did it.

Here are my final settings

```
KEYCLOAK_ADMIN=<...>
KEYCLOAK_ADMIN_PASSWORD=<...>
KC_DB=<...>
KC_DB_URL=<...>
KC_DB_USERNAME=<...>
KC_DB_PASSWORD=<...>
KC_HOSTNAME=qa.myapp.com
KC_FEATURES=token-exchange
KC_PROXY=edge
```

**Reply by aryalireza** [reply]
Oh, that was brilliant, my friend!
This works for me with this docker image version: quay.io/keycloak/keycloak:22.0.5 and these environment variables in the docker-compose file:
```
    Environment:
      - KEYCLOAK_ADMIN=admin
      - KEYCLOAK_ADMIN_PASSWORD=***
      - KC_METRICS_ENABLED=true
      - KC_HOSTNAME=kc.example.com
      - KC_ENABLE_STATISTICS=true
      - KC_HEALTH_ENABLED=true
      - KC_LOG_LEVEL=info
      - KC_PROXY=edge
      - QUARKUS_TRANSACTION_MANAGER_ENABLE_RECOVERY=true
      - KC_HOSTNAME_STRICT=true
      - KC_HTTP_ENABLED=true
      - KC_HTTPS_ENABLED=false
```
